src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,82 @@
-export const Pagination = () => {};
+import React from 'react';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (number: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberOfPages = Math.ceil(total / perPage);
+  const pageNumbers = getNumbers(1, numberOfPages);
+
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === numberOfPages;
+
+  const paginateItems = () => {
+    const allElements = getNumbers(1, total);
+    const startIndex = (currentPage - 1) * perPage;
+    const endIndex = startIndex + perPage;
+
+    return allElements.slice(startIndex, endIndex);
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={`page-item ${isFirstPage ? 'disabled' : ''}`}>
```

Comment:
Use classnames library to set classes conditionally https://www.npmjs.com/package/classnames